sdks/python/apache_beam/runners/dataflow/internal/names.py

```diff
@@ -69,6 +69,7 @@ class PropertyNames(object):
 
   Property strings as they are expected in the CloudWorkflow protos.
   """
+  ALLOWS_SHARDABLE_STATE = 'allows_shardable_state'
```

Comment:
Could you please add some explanations on what `ALLOWS_SHARDABLE_STATE ` and `PRESERVES_KEYS` mean?

Reply:
Done.